crates/gossip/src/protocol.rs

```diff
@@ -0,0 +1,22 @@
+//! Protocol definitions for the gossip module.
+
+use kitsune2_api::{K2Error, K2Result};
+use prost::{bytes, Message};
+
+include!("../proto/gen/kitsune2.gossip.rs");
+
+/// Deserialize a gossip message
+pub fn deserialize_gossip_message(
+    value: bytes::Bytes,
+) -> K2Result<K2GossipProto> {
+    K2GossipProto::decode(value).map_err(K2Error::other)
+}
+
+/// Serialize a gossip message
+pub fn serialize_gossip_message(
+    value: K2GossipProto,
+) -> K2Result<bytes::Bytes> {
+    let mut buf = Vec::new();
+    value.encode(&mut buf).map_err(K2Error::other)?;
+    Ok(bytes::Bytes::from(buf))
```

Comment:
I think at the moment this compiles down to a zero copy since internally bytes is using a Vec... but in the docs it warns not to depend on this behavior because they might switch the internal repr someday. To future proof: https://github.com/holochain/kitsune2/blob/093aceb02791c6ffd206190146bd7803abc2fec6/crates/api/src/protocol.rs#L15-L24

Reply:
Good call, done